src/libsystemd/sd-device/device-enumerator.c

```diff
@@ -273,6 +286,13 @@ _public_ int sd_device_enumerator_allow_uninitialized(sd_device_enumerator *enum
 
         return 1;
 }
+_public_ int sd_device_enumerator_add_all_parents(sd_device_enumerator *enumerator) {
+        assert_return(enumerator, -EINVAL);
+
+        enumerator->parent_match_flags = MATCH_NONE;
```

Comment:
Please reset the `scan_update` flag:
```c
enumerator->scan_uptodate = false;
```
See e.g. `device_enumerator_add_match_is_initialized()` below.